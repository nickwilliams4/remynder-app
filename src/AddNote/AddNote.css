textarea {
  width: 95%;
  background-color: #f5f57e;
}

button {
  height: 2px;
  cursor: pointer;
}

.cancel, .save {
  height: 40px;
  width: 100px;
  border-radius: 5px;
}

.button-icon {
  margin-left: 10px;
  zoom: 1.5;
}
.user-name-field {
  height: 30px;
  width: 20px
}

.AddNote {
  padding-left: 5px;
  background-color: yellow;
  display: flex;
  justify-content: center;
  align-items: center;
}

.frequency-container {
  padding-top: 10px;
  padding-bottom: 10px;
}

@media screen and (min-width: 500px) {
  .AddNote {
    font-size: 25px;
  }
  input[type="text"] {
    font-size: 15px;
  }
  textarea[type="text"] {
    font-size: 20px;
  }
}