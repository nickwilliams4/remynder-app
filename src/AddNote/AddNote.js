import React, { Component } from 'react'
import config from '../config'
import ApiContext from '../ApiContext'
import TokenService from '../services/token-service'
import NotFoundPage from '../NotFoundPage/NotFoundPage'
import './AddNote.css'

const Required = () => (
  <span className='AddNote__required'>*</span>
)

export default class AddNote extends Component {
  static contextType = ApiContext;

  state = {
    error: null,
  };

  handleSubmit = e => {
    e.preventDefault()
    const { title, content, remynder } = e.target
    const notes = {
      title: title.value,
      content: content.value,
      remynder: remynder.value
    }
    this.setState({ error: null })
    fetch(`${config.API_ENDPOINT}/notes`, {
      method: 'POST',
      body: JSON.stringify(notes),
      headers: {
        'Content-Type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }

    })
      .then(res => {
        if (!res.ok) {
          return res.json().then(error => {
            throw error
          })
        }
        return res.json()
      })
      .then(data => {
        this.context.addNote(data)
        this.props.history.push('/NotePage')
      })
      .catch(error => {
        this.setState({ error })
      })
  }
  handleClickCancel = () => {
    this.props.history.push('/NotePage')
  };

  render() {
    const { error } = this.state
    return (
      <NotFoundPage>
        <section className='AddNote'>
          <main role="main">
            <header>
              <h1>New Remynder</h1>
            </header>
            <form
              className="AddNote_form"
              onSubmit={this.handleSubmit}
            >
              <div className='AddNote__error' role='alert'>
                {error && <p>{error.message}</p>}
              </div>
              <div className="form-section">
                <label htmlFor="title">
                  Name of Remynder:
                  {' '}
                </label>
                <Required />
                <input
                  type="text"
                  title="title"
                  id="title"
                  placeholder="Call Jane"
                  required>
                </input>
              </div>
              <div>
                <label htmlFor="content">
                  What's on your mynd?
                  {''}
                  <Required />
                </label>
                <textarea title="content" id="content" rows="15" required></textarea>
              </div>
              <div className="frequency-container">
                <label htmlFor="frequency">How often do you want email remynders? </label>
                <select title="remynder" id="remynder">
                <option value="5">Every 5 Minutes (for testing purposes)</option>
                  <option value="6">Every 6 Hours</option>
                  <option value="12">Every 12 Hours</option>
                  <option value="24">Every 24 Hours</option>
                  <option value="48">Every 2 Days</option>
                </select>
              </div>

              <div className='AddNote__buttons'>
                <button type='button' onClick={this.handleClickCancel}>
                  Cancel
            </button>
                {' '}
                <button type='submit'>
                  Save
            </button>
              </div>
            </form>
          </main>
        </section>
      </NotFoundPage>
    )
  }
}