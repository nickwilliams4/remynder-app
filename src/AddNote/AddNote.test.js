import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddNote from './AddNote'

describe('AddNote component', () => {
  const props = {
    className: 'test-class-name'
  }

  it('renders the AddNote given props', () => {
    const wrapper = shallow(<AddNote {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a form.AddNote by default', () => {
    const wrapper = shallow(<AddNote />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})