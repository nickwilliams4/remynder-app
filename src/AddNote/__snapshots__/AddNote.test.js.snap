// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component renders a form.AddNote by default 1`] = `
<NotFoundPage>
  <section
    className="AddNote"
  >
    <main
      role="main"
    >
      <header>
        <h1>
          New Remynder
        </h1>
      </header>
      <form
        className="AddNote_form"
        onSubmit={[Function]}
      >
        <div
          className="AddNote__error"
          role="alert"
        />
        <div
          className="form-section"
        >
          <label
            htmlFor="title"
          >
            Name of Remynder:
             
          </label>
          <Required />
          <input
            className="user-name-field"
            id="title"
            placeholder="Call Jane"
            required={true}
            title="title"
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="content"
          >
            What's on your mynd?
            <Required />
          </label>
          <textarea
            id="content"
            required={true}
            rows="15"
            title="content"
            type="text"
          />
        </div>
        <div
          className="frequency-container"
        >
          <label
            htmlFor="frequency"
          >
            How often do you want email remynders? 
          </label>
          <select
            id="remynder"
            title="remynder"
          >
            <option
              value="6"
            >
              Every 6 Hours
            </option>
            <option
              value="12"
            >
              Every 12 Hours
            </option>
            <option
              value="24"
            >
              Every 24 Hours
            </option>
            <option
              value="48"
            >
              Every 2 Days
            </option>
          </select>
        </div>
        <div
          className="AddNote__buttons"
        >
          <button
            className="cancel"
            onClick={[Function]}
            type="button"
          >
            Cancel
            <FontAwesomeIcon
              border={false}
              className="button-icon"
              fixedWidth={false}
              flip={null}
              icon="ban"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
           
          <button
            className="save"
            type="submit"
          >
            Save
            <FontAwesomeIcon
              border={false}
              className="button-icon"
              fixedWidth={false}
              flip={null}
              icon="save"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>
      </form>
    </main>
  </section>
</NotFoundPage>
`;

exports[`AddNote component renders the AddNote given props 1`] = `
<NotFoundPage>
  <section
    className="AddNote"
  >
    <main
      role="main"
    >
      <header>
        <h1>
          New Remynder
        </h1>
      </header>
      <form
        className="AddNote_form"
        onSubmit={[Function]}
      >
        <div
          className="AddNote__error"
          role="alert"
        />
        <div
          className="form-section"
        >
          <label
            htmlFor="title"
          >
            Name of Remynder:
             
          </label>
          <Required />
          <input
            className="user-name-field"
            id="title"
            placeholder="Call Jane"
            required={true}
            title="title"
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="content"
          >
            What's on your mynd?
            <Required />
          </label>
          <textarea
            id="content"
            required={true}
            rows="15"
            title="content"
            type="text"
          />
        </div>
        <div
          className="frequency-container"
        >
          <label
            htmlFor="frequency"
          >
            How often do you want email remynders? 
          </label>
          <select
            id="remynder"
            title="remynder"
          >
            <option
              value="6"
            >
              Every 6 Hours
            </option>
            <option
              value="12"
            >
              Every 12 Hours
            </option>
            <option
              value="24"
            >
              Every 24 Hours
            </option>
            <option
              value="48"
            >
              Every 2 Days
            </option>
          </select>
        </div>
        <div
          className="AddNote__buttons"
        >
          <button
            className="cancel"
            onClick={[Function]}
            type="button"
          >
            Cancel
            <FontAwesomeIcon
              border={false}
              className="button-icon"
              fixedWidth={false}
              flip={null}
              icon="ban"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
           
          <button
            className="save"
            type="submit"
          >
            Save
            <FontAwesomeIcon
              border={false}
              className="button-icon"
              fixedWidth={false}
              flip={null}
              icon="save"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>
      </form>
    </main>
  </section>
</NotFoundPage>
`;
